src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourceObject = {};
+
+  sourceString.split(';').forEach((line) => {
+    const trimmedLine = line.trim();
+
+    if (trimmedLine === '') {
+      return;
+    }
+
+    const colonIndex = trimmedLine.indexOf(':'); // Corrected this line
+
+    if (colonIndex === -1) {
+      return;
+    }
+
+    const property = trimmedLine.slice(0, colonIndex).trim();
+    const value = trimmedLine.slice(colonIndex + 1).trim();
+
+    sourceObject[property] = value;
+  });
```

Comment:
Instead of using 'forEach' and modifying the 'sourceObject' directly, consider using 'reduce' to build up the object. This approach aligns with the checklist's recommendation to avoid modifying an existing object within iteration methods.